tests/testthat/test-3agglomerate.R

```diff
@@ -29,137 +29,132 @@ test_that("agglomerate", {
     expect_equal(rd1, rd2)
     # Test that make_unique work
-    uniq <- mergeFeaturesByRank(se, rank = "Species")
-    not_uniq <- mergeFeaturesByRank(se, rank = "Species", make_unique = FALSE)
+    uniq <- agglomerateByRank(se, rank = "Species")
+    not_uniq <- agglomerateByRank(se, rank = "Species", make_unique = FALSE)
     expect_true( !any( duplicated(rownames(uniq)) ) )
     expect_true( any( duplicated(rownames(not_uniq)) ) )
     
+    # Load data
     data(GlobalPatterns, package="mia")
     tse <- GlobalPatterns
-    tse <- transformAssay(tse, assay.type="counts", method="relabundance")
-    altExp(tse, "Family") <- agglomerateByRank(tse, rank="Family")
-    altExp(tse, "Family1") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE)
-    altExp(tse, "Family2") <- agglomerateByRank(tse, rank="Family", onRankOnly = FALSE)
-    altExp(tse, "Family3") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE, na.rm = TRUE)
-    altExp(tse, "Family4") <- agglomerateByRank(tse, rank="Family", onRankOnly = TRUE, na.rm = FALSE)
-    altExp(tse, "Family5") <- agglomerateByVariable(tse, f="Family", MARGIN = 'row')
+
+    # Check that na.rm works
+    # Get all phyla
+    all_phyla <- unique( rowData(tse)$Phylum )
+    
+    # When na.rm = FALSE, then phyla should also include NA --> one extra row
+    test0 <- agglomerateByVariable(tse, MARGIN = 1, f = "Phylum", na.rm = FALSE)
+    test1 <- agglomerateByRank(tse, rank = "Phylum", na.rm = FALSE)
     
-    # Other group is added by agglomerateByPrevalence function to collect features under threshold
-    actual <- agglomerateByPrevalence(tse, rank="Family", assay.type="relabundance", 
-                                         detection  = 0.5/100, prevalence  = 20/100)
-    actual0 <- agglomerateByPrevalence(altExp(tse, "Family"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual1 <- agglomerateByPrevalence(altExp(tse, "Family1"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual2 <- agglomerateByPrevalence(altExp(tse, "Family2"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual3 <- agglomerateByPrevalence(altExp(tse, "Family3"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual4 <- agglomerateByPrevalence(altExp(tse, "Family4"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
-    actual5 <- agglomerateByPrevalence(altExp(tse, "Family5"), assay.type="relabundance", 
-                                       detection  = 0.5/100, prevalence  = 20/100)
+    # Test that dimentionality is the same for merging object by agglomerateByRank
+    # and agglomerateByVariable.
+    expect_equal(nrow(test0), length(all_phyla))
```

Comment:
Good. The add one more check that test that there are more taxa when agglomeration is done to "Species" level for instance.

nrow(agglomerateByRank)) == x

nrow(agglomerateByVariable)) == y

You can hardcode those values. So these checks will inform us if something has changed.